components/ringcentral/sources/common/http-based.js

```diff
@@ -0,0 +1,137 @@
+const template = require("lodash/template");
+const base = require("./base");
+const notificationTypes = require("./notification-types");
+
+module.exports = {
+  ...base,
+  dedupe: "unique",
+  props: {
+    ...base.props,
+    http: {
+      type: "$.interface.http",
+      customResponse: true,
+    },
+  },
+  hooks: {
+    ...base.hooks,
+    async activate() {
+      const opts = {
+        address: this.http.endpoint,
+        eventFilters: this._getEventFilters(),
+      };
+      const {
+        id: webhookId,
+        verificationToken,
+      } = await this.ringcentral.createHook(opts);
+
+      this.db.set("webhookId", webhookId);
+      this.db.set("verificationToken", verificationToken);
+    },
+    async deactivate() {
+      const webhookId = this.db.get("webhookId");
+      await this.ringcentral.deleteHook(webhookId);
+
+      this.db.set("verificationToken", null);
+    },
+  },
+  methods: {
+    ...base.methods,
+    _getPropValues() {
```

Comment:
could you remove this and `lodash/template` in favor of just setting the `filter` attribute to function and use `this` knowing function will be called in the component method? not sure if you think that is too hacky.. not a huge fan of this `_getPropValues`

Reply:
You mean something like this?
```js
// notification-types.js
module.exports = [
  {
    key: "message-event-outbound",
    label: "Message Event (Outbound)",
    filter: ({ extensionId, messageType }) => `/restapi/v1.0/account/~/extension/${extensionId}/message-store?direction=Outbound&type=${messageType}`),
  },
};
```

I'm up to it, with the only caveat that I wanted to keep the `notification-types` as a "data" module without any logic itself (it'd had been a JSON originally if it weren't for the fact that we don't support loading JSON files yet). If you don't see a problem with adding logic to this module then I'm fine with moving the templating logic into that module. To me it's as hacky as the approach I took, so I'm comfortable both ways 🙂

Reply:
i was thinking to use unbound functions so the `this` would be the component when called but I like the concept of keeping that more of a simple data module without functions so let's keep it as is.